### DISCUSSION TITLE
Feature Request - Mutliselect selection should respect order of options

### DISCUSSION THREAD

**Discussion by Kisters-BS** [primefaces/community#479] [PrimeNG] [answered]
As example I am using the demo of: https://primeng.org/multiselect

When I select a option in the middle of the options and then a option which appears before it, it is added at the end of the selection value.
![grafik](https://github.com/primefaces/community/assets/84901454/6fe20c1b-84d0-4520-a447-0b603d81b26e)
![grafik](https://github.com/primefaces/community/assets/84901454/75780a05-4e0f-4a3d-aaeb-61ebd99d5a35)

![grafik](https://github.com/primefaces/community/assets/84901454/b96a31b4-25e8-4359-886f-43637a91b7f0)

I would like that the order of the options is respected in the selection. Maybe add "respectOrder" as an optional property.

I need this feature for https://primeng.org/table#column-toggle, where the columns should be placed at the same position as they are specified in the options.

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Thanks for the feedback, we've added it in our internal backlog.

**Reply by Kisters-BS** [reply]
Thank you for planing this feature!
